src/expr/impl/src/scalar/external/iceberg.rs

```diff
@@ -0,0 +1,197 @@
+// Copyright 2024 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This module contains the expression for computing the iceberg partition value.
+//! spec ref: <https://iceberg.apache.org/spec/#partition-transforms>
+use std::fmt::Formatter;
+use std::str::FromStr;
+use std::sync::Arc;
+
+use anyhow::anyhow;
+use arrow_schema::DataType as ArrowDataType;
+use icelake::types::{
+    create_transform_function, Any as IcelakeDataType, BoxedTransformFunction, Transform,
+};
+use risingwave_common::array::{ArrayRef, DataChunk};
+use risingwave_common::ensure;
+use risingwave_common::row::OwnedRow;
+use risingwave_common::types::{DataType, Datum};
+use risingwave_expr::expr::BoxedExpression;
+use risingwave_expr::{build_function, ExprError, Result};
+
+pub struct IcebergTransform {
+    child: BoxedExpression,
+    transform: BoxedTransformFunction,
+    return_type: DataType,
+}
+
+impl std::fmt::Debug for IcebergTransform {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("IcebergTransform")
+            .field("child", &self.child)
+            .field("return_type", &self.return_type)
+            .finish()
+    }
+}
+
+#[async_trait::async_trait]
+impl risingwave_expr::expr::Expression for IcebergTransform {
+    fn return_type(&self) -> DataType {
+        self.return_type.clone()
+    }
+
+    async fn eval(&self, data_chunk: &DataChunk) -> Result<ArrayRef> {
+        // Get the child array
+        let array = self.child.eval(data_chunk).await?;
+        // Convert to arrow array
+        let arrow_array = array.as_ref().try_into().unwrap();
```

Comment:
Why `unwrap` here?

Reply:
I will convert it as an error in the next PR.